src/main/java/hu/bme/mit/spaceship/TorpedoStore.java

```diff
@@ -30,18 +31,17 @@ public TorpedoStore(int numberOfTorpedos){
 
   public boolean fire(int numberOfTorpedos){
     if(numberOfTorpedos < 1 || numberOfTorpedos > this.torpedoCount){
-      new IllegalArgumentException("numberOfTorpedos");
+      throw new IllegalArgumentException("numberOfTorpedos"); // Fixed #1
     }
 
     boolean success = false;
 
     // simulate random overheating of the launcher bay which prevents firing
-    Random generator = new Random();
     double r = generator.nextDouble();
 
     if (r >= FAILURE_RATE) {
       // successful firing
-      this.torpedoCount =- numberOfTorpedos;
+      this.torpedoCount = this.torpedoCount - numberOfTorpedos; // Fixed #3
```

Comment:
This could have been done with a single '-=' sign. Try to be more effective next time!!!!